### DISCUSSION TITLE
Cart/checkout blocks do not respect filters/actions

### DISCUSSION THREAD

**Discussion by ninetyninew** [woocommerce/woocommerce#45081] [WooCommerce, Blocks, and Site Editing]
Unsure if this is the right repository for cart/checkout blocks - it looks like those are included in this repository, apologies if not the right one (couldn't find another repo).

## Describe the bug

We have read https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/891 and related posts like https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/3978 but just wanted to give some visibility that filters (and potentially actions?) like woocommerce_cart_item_quantity are not used in cart/checkout blocks.

This means the cart/checkout functionality would be different to a user when using a page created from these blocks if a plugin has extended cart/checkout functionality using any standard filters/actions.

Is the plan to include any of the standard filters/actions in all WooCommerce blocks going forward? If not I assume there will be some equivalent in future but seems like this isn't available right now. If that is the case I don't see a way forward for plugins to support blocks at this moment in time if they rely on those filters - and if there was we would need to maintain 2 separate sets of code, one for templates, one for blocks?

Maybe others effected include but not limited to:

- woocommerce_get_item_data
- woocommerce_cart_item_quantity
- 'woocommerce_cart_item_price

Maybe these are effected too?

- woocommerce_add_cart_item_data
- woocommerce_before_calculate_totals (action)

**Comment by Aljullu** [comment]
Hi @ninetyninew, thanks for contacting us! Yes, this is the right repo for the Cart and Checkout blocks. :slightly_smiling_face: 

Extension support in Cart and Checkout blocks is still experimental and we are aware there are some things extensions can tweak in the Cart and Checkout shortcodes which is not possible to tweak in the blocks. Some of them will be enabled in the future while some others won't, it's a bit too early to say because we are still in an experimental phase regarding extensibility.

I'm not able to give any promise about the filters you listed, but I would recommend staying tuned to [developer.woocommerce.com blog](https://developer.woocommerce.com/blog/). You can also check our [extensibility docs](https://github.com/woocommerce/woocommerce-gutenberg-products-block/tree/trunk/docs/extensibility), which for now are focused on payment methods and the Checkout flow, but we will update as more extensibility options are available.

I will close this issue for now since there is nothing directly actionable for us, but thanks for sharing your feedback.
